vertx-template-engines/vertx-web-templ-mvel/src/main/java/io/vertx/ext/web/templ/impl/MVELTemplateEngineImpl.java

```diff
@@ -67,11 +70,16 @@ public void render(RoutingContext context, String templateFileName, Handler<Asyn
       }
       Map<String, RoutingContext> variables = new HashMap<>(1);
       variables.put("context", context);
-      handler.handle(Future.succeededFuture(Buffer.buffer((String)TemplateRuntime.execute(template, variables))));
+      String directoryName = Paths.get(templateFileName).getParent().toAbsolutePath().toString();
```

Comment:
Hi Paulo,
i never used VertxInternal.resolveFile.
MVEL compiler seems to solve correctly, if you set the path in the place where is the template file.
I will try your method this night, in the two situation (jar files or system folder).